compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -564,29 +591,49 @@ class ProtobufGenerator(val params: GeneratorParams) extends DescriptorPimps {
                  |  ${fieldNameSymbol}.foreach($writeFunc)
                  |};""")
           } else if (field.isRequired) {
-          generateWriteSingleValue(field, toBaseType(field)(fieldNameSymbol))(printer)
-        } else if (field.isSingular) {
-          // Singular that are not required are written only if they don't equal their default
-          // value.
-          printer
-            .add(s"{")
-            .indent
-            .add(s"val __v = ${toBaseType(field)(fieldNameSymbol)}")
-            .add(s"if (__v != ${defaultValueForGet(field, uncustomized = true)}) {")
-            .indent
-            .call(generateWriteSingleValue(field, "__v"))
-            .outdent
-            .add("}")
-            .outdent
-            .add("};")
-        } else {
-          printer
-            .add(s"${fieldNameSymbol}.foreach { __v =>")
-            .indent
-            .call(generateWriteSingleValue(field, toBaseType(field)("__v")))
-            .outdent
-            .add("};")
-        }
+            if (field.customSingleScalaTypeName.isDefined) {
```

Comment:
FYI there's an code indentation fix for the subsequent `else if` blocks here so the diff shows a little more code change than needed

Reply:
Would you be able to fix the indentation in a separate commit? It's difficult to see the difference.